src/bufr/BufrParser/Exports/Variables/DatetimeVariable.cpp

```diff
@@ -85,65 +86,98 @@ namespace Ingester
         std::time_t this_time = std::mktime(&tm);
         int64_t diff_time;
 
-        for (unsigned int idx = 0; idx < map.at(yearKey_).size(); idx++)
+        std::vector<int64_t> timeOffsets;
+        timeOffsets.reserve(map.at(getExportKey(ConfKeys::Year))->size());
+
+        for (unsigned int idx = 0; idx < map.at(getExportKey(ConfKeys::Year))->size(); idx++)
         {
+            int year = static_cast<int>(map.at(getExportKey(ConfKeys::Year))->getAsFloat(idx));
+            int month = static_cast<int>(map.at(getExportKey(ConfKeys::Month))->getAsFloat(idx));
+            int day = static_cast<int>(map.at(getExportKey(ConfKeys::Day))->getAsFloat(idx));
+            int hour = static_cast<int>(map.at(getExportKey(ConfKeys::Hour))->getAsFloat(idx));
+            int minutes = 0;
```

Comment:
Why are mins/secs different here? Not 100% sure what this is, but if it is related to offset from UTC to localtime, some timezones are 30 mins off, not on the hour

Reply:
Minutes and seconds are treated differently here because they are optional values where month, day, hour are required. There are BUFR files that don't include information about seconds or minutes... The timestamps aren't always that granular.